pkg/services/sqlstore/dashboard_provisioning.go

```diff
@@ -17,6 +18,23 @@ type DashboardExtras struct {
 	Value       string
 }
 
+func GetProvisionedDataByDashboardId(cmd *models.GetProvisionedDashboardByDashboardId) error {
+	result := &models.DashboardProvisioning{}
+
+	exist, err := x.Where("dashboard_id = ?", cmd.DashboardId).Get(result)
+	if err != nil {
+		return err
+	}
+
+	if !exist {
+		return models.ErrDashboardProvisioningDoesNotExist
```

Comment:
A bit confused. Why not return nil in this case or use the cmd.Result to tell if it's provisioned or not?

Reply:
Using errors for signaling that something expected was not found is quite common in go. 
ex https://github.com/grafana/grafana/blob/master/pkg/services/sqlstore/user.go#L158
So for this query I think it makes sense since it returns the ProvisionedDashboard struct.

We could create a new query to check if it exists or not but I felt that was kinda redundant.